tensorflow/core/kernels/crop_resize_bilinear_core.h

```diff
@@ -0,0 +1,459 @@
+/* Copyright 2015 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef TENSORFLOW_CORE_KERNELS_CROP_RESIZE_BILINEAR_CORE_H_
+#define TENSORFLOW_CORE_KERNELS_CROP_RESIZE_BILINEAR_CORE_H_
+
+namespace tensorflow {
+namespace {
+// Compute the interpolation indices only once.
+struct CachedInterpolation {
+  int lower;  // Lower source index used in the interpolation
+  int upper;  // Upper source index used in the interpolation
+  // 1-D linear iterpolation scale (see:
+  // https://en.wikipedia.org/wiki/Bilinear_interpolation)
+  float lerp;
+};
+
+inline bool compute_single_interpolation_weight(
+    const int in_size, const float out2in_scale, const float out2in_start,
+    const bool clip, const int i, int& lower, int& upper, float& lerp) {
```

Comment:
It's our style guide that out-parameters get passed as pointers and not references (const references are fine since you cannot mutate those)